tests/fixture-valid.js

```diff
@@ -234,6 +362,29 @@ function checkChannel(channel) {
   checkCapabilities(channel, minUsedFineness);
 }
 
+/**
+ * Checks whether the specified string contains only allowed and all required variables
+ * and pushes an error on wrong variable usage. Note that the default of both parameters is an empty array.
+ * @param {!string} str The string to be checked.
+ * @param {!Array.<string>} [requiredVariables=[]] Variables that must be included in the string. Specify them with leading dollar sign ($var).
+ * @param {!Array.<string>} [allowedVariables=[]] Variables that may be included in the string; requiredVariables are automatically included. Specify them with leading dollar sign ($var).
+ */
+function checkTemplateVariables(str, requiredVariables = [], allowedVariables = []) {
```

Comment:
For better clarity, I would not set a default value, requiring all calls to explicitly set required and allowed variables.